src/MPoly.jl

```diff
@@ -1100,6 +1100,17 @@ function _map(g, p::MPolyElem, Rx)
    return finish(M)
 end
 
+###############################################################################
+#
+#   Parent object overload (with coercion)
+#
+###############################################################################
+
+function (S::MPolyRing)(b::Vector{U}, m::Vector{Vector{Int}}) where {T <: RingElement, U <: RingElement}
```

Comment:
```suggestion
function (S::MPolyRing)(b::Vector{U}, m::Vector{Vector{T}}) where {T <: RingElement, U <: RingElement, T<:Union{Int64, UInt64, fmpz}}
```
Can you try this? The Nemo versions have this second argument, that is why they are less specific in the second argument.

Reply:
And remove the first `T` in the `where` clause.

Reply:
Actually T is not currently used in the signature. Maybe that is part of the issue too.

Reply:
And I can't put fmpz in the type in AbstractAlgebra.

Reply:
Works for me if I do `m::Vector{<: Vector}` (no `T`).

Reply:
Yeah I don't see a way to solve this one. In AbstractAlgebra we only allow UInt (whether 32 or 64 bit) for exponents. This is going to conflict with Nemo whatever we do here.

I tried leaving out the fmpz in the signature in AbstractAlgebra and of course it still gives an ambiguity warning.

Obviously if Nemo wants to allow fmpz and Int exponents, it has to implement those itself. But it cannot also then implement functions with UInt64 exponents that AbstractAlgebra already implements.

Reply:
We can just do `m::Vector{<: Vector}`. This works for me.

Reply:
Then whatever exponent types the polynomial ring accepts it will still work.

Reply:
Yeah ok, that should be alright. I'll make that change and push after testing locally.

Reply:
I still get the following locally:

```
nmod_mpoly.exponents: Error During Test at /home/wbhart/.julia/packages/Nemo/j2Z5Q/test/flint/nmod_mpoly-test.jl:796
  Test threw exception
  Expression: S([BigInt(0), BigInt(1), BigInt(2)], m) == 1x + 2y
  MethodError: (::NmodMPolyRing)(::Vector{BigInt}, ::Vector{Vector{BigInt}}) is ambiguous. Candidates:
    (S::MPolyRing{T})(b::Vector{U}, m::Vector{var"#s453"} where var"#s453"<:(Vector{T} where T)) where {T<:RingElement, U<:RingElement} in AbstractAlgebra at /home/wbhart/.julia/dev/AbstractAlgebra/src/MPoly.jl:1109
    (R::NmodMPolyRing)(a::Vector{T} where T, b::Array{Vector{T}, 1}) where T in Nemo at /home/wbhart/.julia/packages/Nemo/j2Z5Q/src/flint/nmod_mpoly.jl:1079
  Possible fix, define
    (::NmodMPolyRing)(::Vector{U}, ::Array{Vector{T}, 1}) where {T, U<:RingElement}
  Stacktrace:
   [1] macro expansion
     @ ~/.julia/packages/Nemo/j2Z5Q/test/flint/nmod_mpoly-test.jl:796 [inlined]
   [2] macro expansion
     @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/.julia/packages/Nemo/j2Z5Q/test/flint/nmod_mpoly-test.jl:725
```

Reply:
What is the exact signature you are using? I can't get it to work.

Reply:
.... even with the latest Nemo.